### PR TITLE
feat: allows assets to be filtered

### PR DESCRIPTION
**Pull request by ryanm101**

### CODE REVIEW

**Comment by ryanm101** [comment]
Make this not required include-assets - pattern "*" for all. 
By default should exclude everything if empty (for now)

File: index.ts [hunk 1]
```diff
@@ -87,6 +89,9 @@ const downloadAssets = async (
     console.log(`Release has ${assets.data.length} assets`);
 
     for (const asset of assets.data) {
+        if (!includeAssets.includes(asset)) {
```
Comment (pcornish) [inline index.ts]: We need to move this check below the `fetchAssetDetails` call.

The `asset` var here just has the `id` but not the name.

We can then use the `assetResponse.name` such as:

```typescript
if (includeAssets && !includeAssets.includes(assetResponse.name)) {
```

or something like this to support regex as well as exact matches:

```typescript
if (includeAssets && !includeAssets.some((ia) => assetResponse.name.match(new RegExp(ia)))) {
```

File: index.ts [hunk 2]
```diff
@@ -62,6 +63,7 @@ async function downloadAsset(
 
 const downloadAssets = async (
     apiKey: string,
+    includeAssets: string[],
```
Comment (pcornish) [inline index.ts]:
```suggestion
    includeAssets: string[] | undefined,
```

File: index.ts [hunk 3]
```diff
@@ -153,10 +158,13 @@ const copyRelease = async () => {
         fs.mkdirSync(tempDir);
     }
 
+    const includeAssets: string[] = INCLUDE_ASSETS.split(" ");
```
Comment (pcornish) [inline index.ts]:
```suggestion
    const includeAssets: string[] = INCLUDE_ASSETS?.split(",");
```

**Comment by ryanm101** [comment]
@pcornish Done, Not sure how to test this though..

**Review by pcornish — APPROVED** [review]

**Pull request merged**